cmd/xl-v1-object.go

```diff
@@ -512,40 +516,96 @@ func (xl xlObjects) PutObject(bucket string, object string, size int64, data io.
 	// Tee reader combines incoming data stream and md5, data read from input stream is written to md5.
 	teeReader := io.TeeReader(limitDataReader, mw)
 
+	// Initialize parts metadata
+	partsMetadata := make([]xlMetaV1, len(xl.storageDisks))
+
 	// Initialize xl meta.
-	xlMeta := newXLMetaV1(object, xl.dataBlocks, xl.parityBlocks)
+	for index := range partsMetadata {
+		partsMetadata[index] = newXLMetaV1(object, xl.dataBlocks, xl.parityBlocks)
```

Comment:
> here each disk will get a different Distribution, make sure that xlMeta.Erasure.Distribution is same for all the disks

It doesn't, Distribution is generated using hashOrder() which returns the same information if we provide the same input which is (object, xl.dataBlocks, xl.parityBlocks) in our case